src/java.base/share/classes/java/lang/invoke/MethodHandleImpl.java

```diff
@@ -2233,6 +2232,29 @@ static Object tableSwitch(int input, MethodHandle defaultCase, CasesHolder holde
         return selectedCase.invokeWithArguments(args);
     }
 
+    // type is validated, value is not
+    static MethodHandle makeConstantReturning(Class<?> type, Object value) {
+        var callType = MethodType.methodType(type);
+        var basicType = BasicType.basicType(type);
+        var form = constantForm(basicType);
+
+        if (type.isPrimitive()) {
```

Comment:
I think you could simplify this using `Wrapper.forBasicType`; all variants should be able to use `wrapper.convert`

Reply:
The cast is not performed if the `type` is an interface; this is a behavioral disparity with the old `value = ptype.cast(value);` in `insertArguments`.

Reply:
Sure, we'd need an explicit `cast` in the `OBJECT` case and that would be a bit redundant. Might still amount to a code simplification.